### PR TITLE
Add support for `NOXPYTHON`, `NOXEXTRAPYTHON` and `NOXFORCEPYTHON`

### LINKED ISSUE #672: Support a `NOXPYTHON` environment variable similar to `NOXSESSION`
### How would this feature be useful?

It'd save a few keystrokes and would work as a default for `--pythons`.

### Describe the solution you'd like

Check for `NOXPYTHON` environment variable:

```shell
NOXPYTHON=3.11 nox -s tests
NOXPYTHON=3.10,3.11 nox -s tests
```

While we're at it, it might make sense to also add env vars for `--extra-pythons` and `--force-pythons`.

### Describe alternatives you've considered

Continue using `--pythons`, or include the Python version in `NOXSESSION`, i.e.

```dotenv
NOXSESSION=tests-3.10
```

but this couples the default session with a Python version

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Closes #672

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ repos:
       - id: trailing-whitespace
 
   - repo: https://github.com/PyCQA/isort
-    rev: 5.10.1
+    rev: 5.12.0
```
Comment (DiddiLeija) [inline .pre-commit-config.yaml:28]: Not sure if `isort` is the only dependency that requires an update. Would like to address that in another thread, though.

Comment (edgarrmondragon) [reply .pre-commit-config.yaml:28]: Yeah, I only update `isort` since it was causing the lint check to fail

Comment (DiddiLeija) [reply .pre-commit-config.yaml:28]: Yup, I've just noticed that. Weird issue though :/

Comment (edgarrmondragon) [reply .pre-commit-config.yaml:28]: @DiddiLeija I've started https://github.com/wntrblm/nox/pull/690 to update all the pre-commit hooks.

Perhaps the pre-commit.ci integration should be added to the repo so `pre-commit autoupdate` is run automatically.

Comment (DiddiLeija) [reply .pre-commit-config.yaml:28]: > Perhaps the pre-commit.ci integration should be added to the repo so `pre-commit autoupdate` is run automatically.

I think we've discussed that before, am I right @wntrblm/nox @henryiii ?

Comment (henryiii) [reply .pre-commit-config.yaml:28]: I'm a big fan of pre-commit.ci.

Poetry-core 1.5.0 broke isort. So you can't build anything before isort 5.12.0 from source. A little ironic given Poetry's stance on SemVer...

I'd like to move over to using Ruff soon, I'll probably make a PR in the near future.

Comment (FollowTheProcess) [reply .pre-commit-config.yaml:28]: Big fan of ruff, would love to see that in Nox!

Comment (edgarrmondragon) [reply .pre-commit-config.yaml:28]: +1 on Ruff 😄

Comment (DiddiLeija) [reply .pre-commit-config.yaml:28]: I don't know a lot about Ruff, but seems like a nice approach! :smile:

File: nox/_options.py [hunk 2]
```diff
@@ -142,11 +142,18 @@ def _envdir_merge_func(
...
-    """Looks at the NOXSESSION env var to set the default value for sessions."""
-    nox_env = os.environ.get("NOXSESSION")
-    env_sessions = nox_env.split(",") if nox_env else None
-    return env_sessions
+def default_env_var_list_factory(env_var: str) -> Callable[[], list[str] | None]:
+    """Looks at the env var to set the default value for a list of env vars.
+
+    Args:
+        env_var (str): The name of the environment variable to look up.
+    """
```
Comment (FollowTheProcess) [inline nox/_options.py:150]: Would ideally like a `Returns` section here, particularly as we're returning a callable which is fine and a good solution for this, just probably not what most people will expect

**Review by FollowTheProcess — CHANGES_REQUESTED** [review]
Nice! Just a small nit around a docstring. Love the sphinx tabs by the way they look great!

File: nox/_options.py (hunk 2 above)
Comment (edgarrmondragon) [reply nox/_options.py]: @FollowTheProcess thanks for the feedback! Done: d1887d5413d6c9d37caa98a455a2243bcc452ec9

**Review by FollowTheProcess — APPROVED** [review]
Nice work, LGTM

**Review by DiddiLeija — APPROVED** [review]
Thank you @edgarrmondragon!

**Comment by edgarrmondragon** [comment]
I think all the feedback's been addressed. Feel free to merge 😄

**Pull request merged**